### ISSUE TITLE
imap tests (especially Timeout) should use a fake clock

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2865]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#2865|
|Type|enhancement|
|Created|2007-11-04 16:45:49Z|
|Branch|https://github.com/twisted/twisted/tree/test-imap-timeout-2865|

This would shorten test time by a few seconds and avoid spurious errors from loaded machines.

<details><summary>Searchable metadata</summary>

```
trac-id__2865 2865
type__enhancement enhancement
reporter__radix radix
priority__highest highest
milestone__ 
branch__branches_test_imap_timeout_2865 branches/test-imap-timeout-2865
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1194194749000000 1194194749000000
changetime__1199280201000000 1199280201000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [21713]) Branching to 'test-imap-timeout-2865'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [21714]) Use a deterministic clock in the tests, and fix the behavior (?)

Refs [#2865](https://github.com/twisted/twisted/issues/2865)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve removed owner|
|-|-|

This is ready to review. There was a problem in test_longFetchDoesntTimeout, because timeout is resetted at login time, so the test wasn't actually testing anything. There may be a strange interaction with the 'blocked' mechanism, but I wasn't able to really understand how it worked.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @therve|
|-|-|

Blech.  The existing tests here are kinda terrible.

Your change looks good.  It seems to interact with the 'blocked' mechanism just fine as far as I can tell.

Hey, look, it's 2008.  Change the copyright statement to reflect that and then merge. ;)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set status to `closed`|
|-|-|

(In [22247]) Merge test-imap-timeout-2865

Author: therve
Reviewer: exarkun
Fixes [#2865](https://github.com/twisted/twisted/issues/2865)

Make the tests of timeout management in imap use a fake clock.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
